app/models/utils/xml_packager.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require "sys/filesystem"
+require "zip"
+require "fileutils"
+
+module Utils
+  # Utility to support a bulk image download operation
+  class XmlPackager < Packager
+    TMP_DIR = "tmp/odk_xml"
+
+    def download_scope
+      responses = Response.accessible_by(@ability, :export)
+      responses = apply_search_scope(responses, @search, @operation.mission) if @search.present?
+
+      responses.joins("INNER JOIN active_storage_attachments ON
+        active_storage_attachments.record_id = responses.id")
+        .joins("INNER JOIN active_storage_blobs ON
+        active_storage_blobs.id = active_storage_attachments.blob_id")
+    end
+
+    def download_and_zip_xml
+      FileUtils.mkdir_p(Rails.root.join(TMP_DIR))
+      responses_ids = download_scope.pluck("id")
+      filename = "#{@operation.mission.compact_name}-xml-responses-"\
+        "#{Time.current.to_s(:filename_datetime)}.zip"
+      zipfile_name = Rails.root.join(TMP_DIR, filename)
+      zip(zipfile_name, responses_ids)
+    end
+
+    def human_readable_xml(response)
+      xml_string = response.odk_xml.download
+      questions = response.form.questionings
```

Comment:
Nit: A question is different from a questioning, let's call this variable `qings` to prevent confusion

Reply:
done!